<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title id='title'>MetaPred2CS at bioinsilico \dot\ org</title>
<link rel="stylesheet" type="text/css" href="/MetaPred2cs/CSS/response.css" />
</head>

<body>
<div class="body_div" >
<div class="header_div" ><div class="title_div" >MetaPred2CS<i> Web server</i></div>
<div class="subtitle_div" ><a href="http://www.bioinsilico.org" >@ bioinsilico \dot\ org</a></div></div><div class="response_div" >
<div style="position:relative;" class="can_pathway" >

  <span class="pathway" ><a href="index.html" >Submit</a></span>
  <span class="pathway" ><a href="#" >Help</a></span>
  <span class="pathway" ><a href="Methods.html" target="_blank" >Methods</a></span>
  <span class="pathway" ><a href="http://metapred2cs.ibers.aber.ac.uk/download/" >Get MetaPred2CS</a></span>
  <span class="pathway" ><a href="disclaimer_page.html" target="_blank" >Disclaimer</a></span>
	
</div>
</span></div>


<div style="text-align:justify;">

<h3_main>

<H3>Submission Form – Pre-computed predictions</H3>

Users are requested to submit the sequences of the query proteins can be either uploaded or pasted in the submission web page (Figure 1).
The format of the sequences must be Fasta and the header should follow the following specifications and format as the example 
shown below:<br><br>

<b> >gi|16128384|ref|NP_414933.1| response regulator in two-component regulatory system with PhoR (or CreC) [Escherichia coli str. K-12 substr. MG1655] <br>
</b> <br>
The different fields of the header are delimited by the pipe character “|” and include the NCBI gi and RefSEQ identification numbers, 
a short description and the Source organism. This information is required during the prediction process, as they are used to query 
different databases. The repository of NCBI annotated prokaryotes genomes including FASTA files in the required format can be downloaded 
from <a href="ftp://ftp.ncbi.nih.gov/genomes/archive/old_refseq/Bacteria/" target="_blank">here</a>.<br>

<p align="center"><img src="/MetaPred2cs/images/y1.png" width="75%"/></p>
<p align="center"><b>Figure 1.</b> Submission web page.</p> <br>

On the first instance, MetaPred2CS submit the sequence to a database of pre-computed predictions. 
If the prediction for the given pair of two component proteins, MetaPred2CS returns a table showing the gene identification 
of the pair of submitted sequences (locus tag) and the prediction score (Figure2). The prediction score ranges between 0 and 1 
and the highest the score, the more likely that the two proteins are part of same Histidine Kinases – Response regulator pair. 
Based on the benchmarking results, predictions can be classified as <b> high confidence, medium confidence </b> or <b> low confidence </b> if  
the score ranges between <b> 0.999-0.700, 0.699-0.400 or 0.399-0.000, </b> respectively.

<p align="center"><img src="/MetaPred2cs/images/6.png" width="75%"/></p>
<p align="center"><b>Figure 2.</b> Example of result table from pre-computed predictions.</p><br>

If the given pair of proteins is not present in the pre-computed database, the user is directed to a second submission web page. 

<H3>Submission Form – De novo predictions</H3>

Besides the sequences of the query proteins, users need to set the prediction 
parameters if not agree with default values.
Information about these parameters and individual predictors that employed during the development of this metapredictor
can be found in <b>Methods page</b>. Method page can be reached from the top menu.
Note that the sequences are inherited from the first submission web page. Structure of a result page which will be 
generated by MetaPred2CS for a <i>de novo</i> predictions can be seen below.<br> 



<h4>Results web page</h4>

Upon completion, a results web-page will be displayed (Figure 3). 
This page is composed of four different tables. The first three tables describe the information on the submitted files 
and prediction parameters, e.g. cut-off values for each individual preidciton methods. The final table contains the prediction score, 
that ranges between 0-1 Based on prediction score, predictions can be classified as <b> high confidence, medium confidence </b> or 
<b> low confidence </b>. Score ranges of these classes are defined as <b> 0.999-0.700, 0.699-0.400 and 0.399-0.000 </b>, respectively.

<p align="center"><img src="/MetaPred2cs/images/4.png" width="75%"/></p>
<p align="center"><b>Figure 3.</b> Results web page.</p><br><br>

<div class="footer_div" >
<hr/><h5 >Finding it useful? Please cite the server.</h5>
</div>
</div>

</body>
</html>
