<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta content="MetaPred2cs: prediction of two-component system protein protein interactions" name="description">
<title id='title'>MetaPred2CS Web server at bioinsilico \dot\ org</title>
<link href="/MetaPred2cs/CSS/response.css" rel="stylesheet" type="text/css">
</head>

<body>
<div class="body_div" >
 <div class="header_div" ><div class="title_div" >MetaPred2CS<i> Web server</i></div>
<div class="subtitle_div" ><a href="http://www.bioinsilico.org" >@ bioinsilico \dot\ org</a></div></div><div class="response_div" >

  <div style="position:relative;" class="can_pathway" >
  <span class="pathway" ><a href="index.html" >Submit</a></span>
  <span class="pathway" ><a href="help_page1.html" target="_blank" >Help</a></span>
  <span class="pathway" ><a href="#" target="_blank" >Methods</a></span>
  <span class="pathway" ><a href="http://metapred2cs.ibers.aber.ac.uk/download/" >Get MetaPred2CS</a></span>
  <span class="pathway" ><a href="disclaimer_page.html" target="_blank" >Disclaimer</a></span>

</div>
<div style="text-align: justify;">

<style>
textarea {
    resize: none;
}
</style>


<h3_main>
  <H3> General Information </H3>

MetaPred2CS is a meta-predictor specifically designed to predict interactions in prokaryotic two-component system,
i.e the pairing between histidine kinases and response regulators.
MetaPred2CS is based on <a href="http://www.ncbi.nlm.nih.gov/pubmed/17160063" target="_blank">Support Vector Machine (SVM)
</a>
that combines six individual sequence based protein-protein interaction prediction
methods: two co-evolutionary based methods (<a href="http://www.ncbi.nlm.nih.gov/pubmed/11933068" target="_blank">in-silico
two hybrid (i2H)</a> and <a href="http://www.ncbi.nlm.nih.gov/pubmed/11707606" target="_blank">mirror tree (MT)
</a> methods) and four genomic context
methods (<a href="http://www.ncbi.nlm.nih.gov/pubmed/15947018" target="_blank">phylogenetic profiling (PP)</a>,
<a href="http://www.ncbi.nlm.nih.gov/pubmed/10573422" target="_blank">gene fusion (GF)</a>,
<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.0030043" target="_blank">
gene neighbourhood (GN)</a>
and <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.0030043" target="_blank">
gene operon methods (GO)</a>).
<br><br>

All methods implemented in MetaPred2CS require a <a href="http://www.ncbi.nlm.nih.gov/pubmed/2231712" target="_blank">BLAST
</a> search to
identify homologous proteins to the query proteins. In the case of i2H and MT methods, the search is perfomed against
<a href="http://www.uniprot.org/downloads" target="_blank">UniProtKB</a> downloaded
from UniProt. In the case of genomics context based methods, sequences search are performed against our local reference
genome database that included 243 genomes (see Table 1). The genomes and genome annotation were downloaded
from <a href="ftp://ftp.ncbi.nih.gov/genomes/archive/old_refseq/Bacteria/" target="_blank">NCBI database</a>
and operon architecture and transcription
units from <a href="http://microbiome.wlu.ca/public/TUpredictions/Predictions/" target="_blank"> here</a>.<br><br>

Lastly, Metapred2CS has been trained in single-domain Histidine kinase and Response regulator proteins,
in the case of <b> hybrid </b> two-component system proteins, users need to decompose the protein sequence
in separate domains
and submit them separately. <br><br>

<script>
function toggleTable() {
    var lTable = document.getElementById("loginTable");
    lTable.style.display = (lTable.style.display == "table") ? "none" : "table";
}
</script>
<style>


table {
    width: 70%;
    margin-left: auto;
    margin-right: auto;
    font-family:sans-serif;
    font-size:10pt;
}

th {
  font-weight: bold;
  font-family:sans-serif;
  font-size:10pt;
  background: #CEDADA;
                text-align:center;
                }
td {
  font-family:sans-serif;
  font-size:10pt;
  background: #F0F0F0;
  text-align:center;
}

</style>
<a id="loginLink" onclick="toggleTable();" href="#">Table 1</a>.
<table id="loginTable" border="1" align="center" style="display:none">

<tr><th>Class/Group</th>
<th>Organism Number Included in Database</th></tr>
<tr><td>Alphaproteobacteria</td><td>27</td></tr>
<tr><td>Aquificae</td><td>1</td></tr>
<tr><td>Firmicutes</td><td>49</td></tr>
<tr><td>Bacteroidetes/Chlorobi</td><td>5</td>
</tr><tr><td>Delta/epsilon subdivisions</td><td>10</td>
</tr><tr><td>Actinobacteria</td><td>16</td>
</tr><tr><td>Betaproteobacteria</td><td>13</td>
</tr><tr><td>Spirochaetes</td><td>2</td>
</tr><tr><td>Gammaproteobacteria</td><td>49</td>
</tr><tr><td>Chlamydiae/Verrucomicrobia</td><td>10</td>
</tr><tr><td>Deinococcus-Thermus</td><td>3</td>
</tr><tr><td>Fusobacteria</td><td>1</td></tr>
<tr><td>Cyanobacteria</td><td>11</td></tr>
<tr><td>Fibrobacteres/Acidobacteria</td><td>3</td></tr>
<tr><td>Spirochaetes</td><td>4</td></tr>
<tr><td>Tenericutes</td><td>12</td></tr>
<tr><td>Planctomycetes</td><td>1</td></tr>
<tr><td>Thermotogae</td><td>1</td></tr>
<tr><td>Chloroflexi</td><td>2</td></tr>
<tr><td>Crenarchaeota*</td><td>6</td></tr>
<tr><td>Euryarchaeota*</td><td>16</td></tr>
<tr><td>Nanoarchaeota*</td><td>1</td></tr>
</table>
<br>

<br><b> Training & Test - The P+ and P- datasets </b><br><br>
The <a href="P+_dataset.faa" target="_blank">P+</a> and <a href="P-_dataset.faa" target="_blank">P-</a>
datasets contain 113 interacting
and 1134 non-interacting experimentally validated TCS pairs respectively,
and were compiled and manually curated from the current literature. These datasets and different sub-classes of these sets
were used to train and test the MetaPred2CS using a k-fold cross validation strategy.


<h4>Prediction Parameters</h4>
<b>i2H and MT methods </b><br><br>

At the second section of submission form users have the option of tune the default values of the six different
parameters used by i2H and MT methods (Figure 1). <br>

<p align="center"><img src="/MetaPred2cs/images/y3.png" width="75%"/></p>
<p align="center"><b>Figure 1.</b> Parameter i2H and Mirror tree methods.</p><br>

i2H and MT methods requires multiple sequence alignments  that are created automatically from user's
query proteins using BLAST and <a href="http://www.ncbi.nlm.nih.gov/pubmed/17846036" target="_blank">clustalw</a>.
For BLAST search, users can adjust the <a href="http://www.ncbi.nlm.nih.gov/BLAST/tutorial/" target="_blank">
e-value</a>
and <a href="http://www.ncbi.nlm.nih.gov/BLAST/tutorial/" target="_blank">b_value</a>.
The rest of the parameters will be used as default such as the as scoring matrix,
which is <a href="http://www.ncbi.nlm.nih.gov/Class/FieldGuide/BLOSUM62.txt" target="_blank">BLOSUM_62</a>.

The <b> Min Matches </b> and <b> Max Matches </b> parameters stand for the minimum and maximum number of
common species in both multiple
sequence alignments between both query proteins. These parameters are used to filter the number of sequence and reduce
the number of them. If the minimum number of common species is not reached, the job will be terminated.
In this case, users need to re-submit using  lower cut-off value for the Min Matches paremeter and/or a higher
b_value for the BLAST search.
The final set of parameters is the correlation function and the substitution matrix.
The available options for the scoring function are: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11933068" target="_blank">
corr </a> and
<a href="http://www.ncbi.nlm.nih.gov/pubmed/11933068" target="_blank"> rankorr</a> and
the substitution matrice:
<a href="http://csbg.cnb.csic.es/pazos/Xdet/Maxhom_GCG.metric" target="_blank">
Maxhom_GCG.metric</a>
and <a href="http://csbg.cnb.csic.es/pazos/Xdet/Maxhom_McLachlan.metric" target="_blank">
Maxhom_McLachlan.metric</a>.<br><br>

<b> Genomic context methods </b> <br><br>
At third section of submission form users has de option of tune the default parameters in the genome methods (Figure 2).
<br>

<p align="center"><img src="/MetaPred2cs/images/y4.png" width="75%"/></p>
<p align="center"><b>Figure 2.</b> Paramers Genomic context methods.</p><br>

In the case of Phylogenetic profiles users can change the E-value cut-off of  the BLASTP search used to detect
homologue proteins across the reference. This parameter is very important as discussed in
this <a href="http://www.ncbi.nlm.nih.gov/pubmed/15947018" target="_blank">work</a>.
The second parameter accounts for <a href="http://www.ncbi.nlm.nih.gov/pubmed/15947018" target="_blank">
mutual information</a>
to infer functional linkages between proteins.

Likewise, in the case of Gene Fusion methods, users can also change the E-value cut-off on BLASTP . Moreover, users can
tune the Bits cut-off on the local alignments performed using
<a href="http://computing.bio.cam.ac.uk/local/doc/fasta_guide.pdf" target="_blank">ssearch36</a>
for query proteins which using <a href="http://www.ncbi.nlm.nih.gov/pubmed/1774068" target="_blank">
Smith-Waterman algorithm</a> to identify fusion events.
Finally, in the case Gene Neighbourhood & Gene Operon methods, the two tunable parameters are the E-value cut-off on the
BLASTP search and the cut-off genomic distance. The latter is important to define neighbouring genes and have been
extensive discussed in the following works:
<a href="http://www.ncbi.nlm.nih.gov/pubmed/10823905" target="_blank">Salgado H. et al., 2000</a>;
<a href="http://www.ncbi.nlm.nih.gov/pubmed/12952538" target="_blank">Strong M. et al., 2003</a>;
<a href="http://www.ncbi.nlm.nih.gov/pubmed/11222772" target="_blank">Ermolaeva MD. et al, 2001</a>;
<a href="http://www.ncbi.nlm.nih.gov/pubmed/12169563" target="_blank">Moreno-HG. et al., 2002</a>;
<a href="http://www.ncbi.nlm.nih.gov/pubmed/10077608" target="_blank">Ross O. et al., 1999</a>
being 200bp (default) the generally accepted cut-off value.

